From 9fc363f03fb69db8bf25a6c854489d58eb11617e Mon Sep 17 00:00:00 2001
From: Lukas Nykryn <lnykryn@redhat.com>
Date: Fri, 9 Feb 2024 15:51:36 +0100
Subject: [PATCH] man: reorder the list of supported naming schemes

Let's put the upstream version first, followed by the RHEL9 backports
and move RHEL8 ones to separate section.

rhel-only

Related: RHEL-7026
---
 man/systemd.net-naming-scheme.xml | 104 +++++++++++++-----------------
 1 file changed, 46 insertions(+), 58 deletions(-)

diff --git a/man/systemd.net-naming-scheme.xml b/man/systemd.net-naming-scheme.xml
index 532d6ddb15..a5903c6d04 100644
--- a/man/systemd.net-naming-scheme.xml
+++ b/man/systemd.net-naming-scheme.xml
@@ -460,26 +460,58 @@
         </varlistentry>
 
         <varlistentry>
-          <term><constant>rhel-8.0</constant></term>
+          <term><constant>rhel-9.0</constant></term>
 
-          <listitem><para>Naming was changed for virtual network interfaces created with SR-IOV and NPAR and
-          for devices where the PCI network controller device does not have a slot number associated.</para>
+          <listitem><para>Since version <constant>v247</constant> we no longer set
+          <varname>ID_NET_NAME_SLOT</varname> if we detect that a PCI device associated with a slot is a PCI
+          bridge as that would create naming conflict when there are more child devices on that bridge. Now,
+          this is relaxed and we will use slot information to generate the name based on it but only if
+          the PCI device has multiple functions. This is safe because distinct function number is a part of
+          the device name for multifunction devices.</para>
+          </listitem>
+        </varlistentry>
 
-          <para>SR-IOV virtual devices are named based on the name of the parent interface, with a suffix of
-          <literal>v<replaceable>port</replaceable></literal>, where <replaceable>port</replaceable> is the
-          virtual device number. Previously those virtual devices were named as if completely independent.
-          </para>
+        <varlistentry>
+           <term><constant>rhel-9.1</constant></term>
 
-          <para>The ninth and later NPAR virtual devices are named following the scheme used for the first
-          eight NPAR partitions. Previously those devices were not renamed and the kernel default
-          ("eth<replaceable>N</replaceable>") was used.</para>
+           <listitem><para>Same as naming scheme <constant>rhel-9.0</constant>.</para></listitem>
+        </varlistentry>
 
-          <para>Names are also generated for PCI devices where the PCI network controller device does not
-          have an associated slot number itself, but one of its parents does. Previously those devices were
-          not renamed and the kernel default was used.</para>
+        <varlistentry>
+           <term><constant>rhel-9.2</constant></term>
+
+           <listitem><para>Same as naming scheme <constant>rhel-9.0</constant>.</para></listitem>
+        </varlistentry>
+
+        <varlistentry>
+            <term><constant>rhel-9.3</constant></term>
+
+          <listitem><para>Naming was changed for SR-IOV virtual device representors.</para>
+
+          <para>The <literal>r<replaceable>slot</replaceable></literal> suffix was added to differentiate SR-IOV
+          virtual device representors attached to a single physical device interface.
+          </para>
           </listitem>
         </varlistentry>
 
+        <varlistentry>
+           <term><constant>rhel-9.4</constant></term>
+
+           <listitem><para>Same as naming scheme <constant>rhel-9.3</constant>.</para></listitem>
+        </varlistentry>
+      </variablelist>
+
+    <para>By default <constant>rhel-9.0</constant> is used.</para>
+    <refsect2>
+      <title>RHEL-8 schemes</title>
+      <para>It is also possible to pick a scheme from RHEL-8</para>
+      <variablelist>
+        <varlistentry>
+          <term><constant>rhel-8.0</constant></term>
+
+          <listitem><para>Same as naming scheme <constant>v239</constant>.</para></listitem>
+        </varlistentry>
+
         <varlistentry>
           <term><constant>rhel-8.1</constant></term>
 
@@ -552,52 +584,8 @@
 
           <listitem><para>Same as naming scheme <constant>rhel-8.7</constant>.</para></listitem>
         </varlistentry>
-
-        <varlistentry>
-          <term><constant>rhel-9.0</constant></term>
-
-          <listitem><para>Since version <constant>v247</constant> we no longer set
-          <varname>ID_NET_NAME_SLOT</varname> if we detect that a PCI device associated with a slot is a PCI
-          bridge as that would create naming conflict when there are more child devices on that bridge. Now,
-          this is relaxed and we will use slot information to generate the name based on it but only if
-          the PCI device has multiple functions. This is safe because distinct function number is a part of
-          the device name for multifunction devices.</para>
-          </listitem>
-        </varlistentry>
-
-        <varlistentry>
-           <term><constant>rhel-9.1</constant></term>
-
-           <listitem><para>Same as naming scheme <constant>rhel-9.0</constant>.</para></listitem>
-        </varlistentry>
-
-        <varlistentry>
-           <term><constant>rhel-9.2</constant></term>
-
-           <listitem><para>Same as naming scheme <constant>rhel-9.0</constant>.</para></listitem>
-        </varlistentry>
-
-        <varlistentry>
-            <term><constant>rhel-9.3</constant></term>
-
-          <listitem><para>Naming was changed for SR-IOV virtual device representors.</para>
-
-          <para>The <literal>r<replaceable>slot</replaceable></literal> suffix was added to differentiate SR-IOV
-          virtual device representors attached to a single physical device interface.
-          </para>
-          </listitem>
-        </varlistentry>
-
-        <varlistentry>
-           <term><constant>rhel-9.4</constant></term>
-
-           <listitem><para>Same as naming scheme <constant>rhel-9.3</constant>.</para></listitem>
-        </varlistentry>
-
       </variablelist>
-
-    <para>Note that <constant>latest</constant> may be used to denote the latest scheme known (to this
-    particular version of systemd).</para>
+    </refsect2>
   </refsect1>
 
   <refsect1>
